### PR TITLE
fix(types): add instance.dataValues property to model.d.ts

### PR DESCRIPTION
**Pull request by spearmootz**
this allows typescript to have a definition for the instance.dataValues property

## Pull Request Checklist

- [ ] Have you added new tests to prevent regressions?
- [x] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

this allows to get typings from 

const sampleInstance = SampleModel.findByPk(pk);
sampleInstance.dataValues; //This currently does not exist in typings

## Todos

- [ ] 
- [ ] 
- [ ]

### CODE REVIEW

File: src/model.d.ts [hunk 1]
```diff
@@ -2093,6 +2093,11 @@ export abstract class Model<TModelAttributes extends {} = any, TCreationAttribut
    */
   _attributes: TModelAttributes; // TODO [>6]: make this a non-exported symbol (same as the one in hooks.d.ts)
 
+  /**
+   * Object that contains undelying model data
```
Comment (WikiRik) [inline src/model.d.ts:2097]:
```suggestion
   * Object that contains underlying model data
```

**Review by WikiRik — COMMENTED** [review]
We'll take a closer look at this later but you can add [this commit](https://github.com/sequelize/sequelize/commit/eada631cc86e3ff4000230a7459011b138e518f8) to the PR to have this change supported by at least one test

File: src/model.d.ts (hunk 1 above)
Comment (spearmootz) [reply src/model.d.ts]: @WikiRik thank you for taking a look.  i have made the suggested change and have pushed the comment change and the adding of the test.

File: test/unit/model/remove-attribute.test.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { DataTypes } from '@sequelize/core';
+import { expect } from 'chai';
+import size from 'lodash/size';
+import { getTestDialectTeaser, sequelize } from '../../support';
+
+describe(getTestDialectTeaser('Model'), () => {
```
Comment (ephys) [inline test/unit/model/remove-attribute.test.ts:6]: looks like you migrated remove-attribute.test.js to TypeScript. I'm thankful for it but don't forget to delete the existing remove-attribute.test.js test

**Review by ephys — CHANGES_REQUESTED** [review]
Other than deleting remove-attribute.test.js, this should be good to go I think

**Review by ephys — APPROVED** [review]

File: test/unit/model/overwriting-builtins.test.js [hunk 3]
```diff
@@ -1,23 +0,0 @@
-'use strict';
-
-const chai = require('chai');
-
-const expect = chai.expect;
-const Support = require('../../support');
-const { DataTypes } = require('@sequelize/core');
-
-describe(Support.getTestDialectTeaser('Model'), () => {
```
Comment (ephys) [inline test/unit/model/overwriting-builtins.test.js:9]: this was accidentally removed

**Review by ephys — CHANGES_REQUESTED** [review]

File: test/unit/model/overwriting-builtins.test.js (hunk 3 above)
Comment (spearmootz) [reply test/unit/model/overwriting-builtins.test.js:9]: Yes, reverting now

Comment (spearmootz) [reply test/unit/model/overwriting-builtins.test.js]: Ok, it's reverted. Sorry about that

**Review by WikiRik — APPROVED** [review]

**Comment by spearmootz** [comment]
Just wondering, will this make it to v6? Or do I have to make the PR elsewhere?

**Comment by WikiRik** [comment]
> Just wondering, will this make it to v6? Or do I have to make the PR elsewhere?

Not directly, but feel free to open a new PR towards v6. Just the change to model.d.ts wil do fine, no need to update the test

**Comment by spearmootz** [comment]
@WikiRik if you cloud. its the same change

https://github.com/sequelize/sequelize/pull/15240

**Pull request merged**